openshift/hack/generate_requirements.py

```
@@ -0,0 +1,1259 @@
+            )
+            ok, _ = _pip_compile(test_in, test_out, ["--allow-unsafe"])
+            if ok:
+                auto_iso_pins[norm] = ver
+                print(f"    ✓ build-isolation pin: {norm}=={ver}")
+            # If not ok: conflicts with pre-build constraints — skip
+
+    pre_build_pin_norms: dict[str, str] = {**auto_iso_pins}
+
+    if auto_iso_pins:
+        print(
+            f"  Auto-detected build-isolation pins for pre-build: "
```

Ruff (0.15.21)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)